src/RageSoundReader_Chain.cpp

```diff
@@ -186,19 +186,19 @@ void RageSoundReader_Chain::Finish()
 	m_iActualSampleRate = GetSampleRateInternal();
 	if( m_iActualSampleRate == -1 )
 	{
-		for (RageSoundReader *it : m_apLoadedSounds)
+		for (auto &pSound : m_apLoadedSounds)

```

Comment:
Prefer specifying types. In this case it'd be:

`for (RageSoundReader*& pSound : m_apLoadedSounds)`

If `*&` is ugly, could consider using a normal for loop via:
```cpp
for (size_t i = 0; i < m_apLoadedSounds.size(); ++i)
{
    RageSoundReader* old = m_apLoadedSounds[i];
    m_apLoadedSounds[i] =
        new RageSoundReader_Resample_Good(old, m_iPreferredSampleRate);
}
```